File: WorkerServiceVB/ActionService.vb
Imports Serilog

Public Class ActionService
    Private _lastRun As Date = Date.Now

    Public Sub Run(action As String, Optional delay As Integer = 5)
        If Date.Now.Subtract(_lastRun).TotalSeconds > delay Then
            _lastRun = Date.Now
            Log.Information($"{action} called at {_lastRun}")
        End If
    End Sub
End Class

File: WorkerService/ActionService.cs
namespace WorkerService;
public class ActionService
{
    private DateTime _lastRun = DateTime.Now;

    public void Run(string action, int delay = 5)
    {
        if (DateTime.Now.Subtract(_lastRun).TotalSeconds > delay)
        {
            _lastRun = DateTime.Now;
            Console.WriteLine($"{action} called at {_lastRun}");
        }
    }
}